src/converters/downcast.js

```diff
@@ -0,0 +1,492 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module table/converters/downcast
+ */
+
+import ViewPosition from '@ckeditor/ckeditor5-engine/src/view/position';
+import ViewRange from '@ckeditor/ckeditor5-engine/src/view/range';
+import TableWalker from './../tablewalker';
+import { toWidget, toWidgetEditable } from '@ckeditor/ckeditor5-widget/src/utils';
+
+/**
+ * Model table element to view table element conversion helper.
+ *
+ * This conversion helper creates whole table element with child elements.
+ *
+ * @param {Object} options
+ * @param {Boolean} options.asWidget If set to true the downcast conversion will produce widget.
+ * @returns {Function} Conversion helper.
+ */
+export function downcastInsertTable( options = {} ) {
+	return dispatcher => dispatcher.on( 'insert:table', ( evt, data, conversionApi ) => {
+		const table = data.item;
+
+		if ( !conversionApi.consumable.consume( table, 'insert' ) ) {
+			return;
+		}
+
+		// Consume attributes if present to not fire attribute change downcast
+		conversionApi.consumable.consume( table, 'attribute:headingRows:table' );
+		conversionApi.consumable.consume( table, 'attribute:headingColumns:table' );
+
+		const asWidget = options && options.asWidget;
+
+		const tableElement = conversionApi.writer.createContainerElement( 'table' );
+
+		let tableWidget;
+
+		if ( asWidget ) {
+			tableWidget = toWidget( tableElement, conversionApi.writer );
+		}
+
+		const tableWalker = new TableWalker( table );
+
+		const tableAttributes = {
+			headingRows: parseInt( table.getAttribute( 'headingRows' ) || 0 ),
```

Comment:
Is `parseInt` really needed here? Aren't attributes in model already numbers (or `undefined`)?

Reply:
Weren't there some problems with strings? I don't recall how it was working but I've managed to have `colspan` or other attributes as: `"11111"` in some converters.

Reply:
No can do: https://github.com/ckeditor/ckeditor5-engine/issues/1420.

Reply:
I'm using a default converters here so the values are strings.

Reply:
Actually I've fixed this for `headingRows` & `headingColumns` - so table attributes. The `colspan` and `rowspan` parsing must use `parseInt()` for now.